barretenberg/cpp/src/barretenberg/relations/utils.hpp

```diff
@@ -208,7 +210,49 @@ template <typename Flavor> class RelationUtils {
     }
 
     /**
-     * @brief Scale elements by consecutive powers of the challenge then sum
+     * @brief Scales elements, representing evaluations of polynomials in subrelations, by separate challenges and then
+     * sum them together. This function has identical functionality with the one above with the caveat that one such
+     * evaluation is part of a linearly dependent subrelation and hence needs to be accumulated separately.
+     *
+     * @details Such functionality is needed when computing the evaluation of the full relation at a specific row in
+     * the execution trace because a linearly dependent subrelation does not act on a specific row but rather on the
+     * entire execution trace.
+     *
+     * @param tuple
+     * @param challenges
+     * @param current_scalar
+     * @param result
+     * @param linearly_dependent_contribution
+     */
+    static void scale_and_batch_elements_without_linear_contributions(auto& tuple,
```

Comment:
i'm sure there is a better way to do this stuff in c++, very open to suggestions and a follow-up refactoring PR :)